src/common/snippets/src/pass/hash.cpp

```diff
@@ -0,0 +1,487 @@
+    // and Results. So we need to put them into sorted_ops separately to ensure correct order of inputs and outputs.
+    {
+        std::vector<std::shared_ptr<ov::Node>> result;
+        result.reserve(sorted_ops.size());
+        for (const auto& param : model.get_parameters()) {
+            result.emplace_back(param);
+        }
+        for (auto&& node : sorted_ops) {
+            if (!ov::op::util::is_parameter(node) && !ov::op::util::is_output(node) && !ov::op::util::is_sink(node))
+                result.emplace_back(node);
+        }
+        for (const auto& sink : model.get_sinks()) {
+            result.emplace_back(sink);
+        }
+        for (const auto& res : model.get_results()) {
+            result.emplace_back(res);
+        }
+        sorted_ops = result;
+    }
+
+    for (const auto& n : sorted_ops) {
+        ov::Node* node = n.get();
+        const std::string& node_type_name{node->get_type_name()};
+
+        OPENVINO_ASSERT(layer_ids.find(node) != layer_ids.end(), "Internal error");
+        // <layers>
+        hash = hash_combine(hash, std::string("layer"));
+        hash = hash_combine(hash_combine(hash, std::string("id")), layer_ids.find(node)->second);
+        hash = hash_combine(hash_combine(hash, std::string("type")), node_type_name);
+
+        // <layers/data> general attributes
+        hash = hash_combine(hash, std::string("data"));
+        auto append_runtime_info = [&](uint64_t& hash, ov::RTMap& attributes) {
+            hash = hash_combine(hash, std::string("rt_info"));
+            for (auto& item : attributes) {
+                if (item.second.is<ov::RuntimeAttribute>()) {
+                    auto& rt_attribute = item.second.as<ov::RuntimeAttribute>();
+                    const auto& type_info = rt_attribute.get_type_info();
+                    if (!strcmp(type_info.name, "fused_names")) {
+                        continue;
+                    }
+                    hash = hash_combine(hash, std::string("attribute"));
+                    hash = hash_combine(hash_combine(hash, std::string("name")), type_info.name);
+                    hash = hash_combine(hash_combine(hash, std::string("version")), type_info.get_version());
+
+                    rt_info::RTInfoHasher rt_info_visitor(hash);
+                    rt_attribute.visit_attributes(rt_info_visitor);
+                }
+            }
+        };
+
+        append_runtime_info(hash, node->get_rt_info());
+
+        int port_id = 0;
+        // <layers/input>
+        if (node->get_input_size() > 0) {
+            hash = hash_combine(hash, std::string("input"));
+            for (auto& i : node->inputs()) {
+                hash = hash_combine(hash, std::string("port"));
+                hash = hash_combine(hash_combine(hash, std::string("id")), port_id++);
+                hash = hash_combine(hash_combine(hash, std::string("precision")), get_precision_name(i.get_element_type()));
+                for (auto d : i.get_partial_shape()) {
```

Comment:
`PartialShape` has [to_string()](https://github.com/openvinotoolkit/openvino/blob/f4fe8400a71077491fef94beafc5e659cae1a3fb/src/core/include/openvino/core/partial_shape.hpp#L355) method, so we don't have to convert it to string manually here.

Here, and also on L429-435

Reply:
nice,  `PartialShape` `to_string()` is used, thanks Ivan!